Tasks/KubernetesV1/src/kubernetes.ts

```diff
@@ -82,14 +99,65 @@ function executeKubectlCommand(clusterConnection: ClusterConnection, command: st
         "TaskEndpointId",
         "KubernetesV1",
         JSON.stringify(telemetry));
-    var result = "";
-    return commandImplementation.run(clusterConnection, command, (data) => result += data)
-    .fin(function cleanup() {
-        var commandOutputLength = result.length;
-        if (commandOutputLength > environmentVariableMaximumSize) {
-            tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
-        } else {
-            tl.setVariable('KubectlOutput', result);
-        }
-    });
+
+    // The output result can contain more than one Json objects
+    // We want to parse each of the objects separately, hence push the output in JSON array form    
+    var result = [];
+    return commandImplementation.run(clusterConnection, command, (data) => result.push(data))
+        .fin(function cleanup() {
+            console.log("commandOutput" + result);
+            var commandOutputLength = result.length;
+            if (commandOutputLength > environmentVariableMaximumSize) {
+                tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
+            } else {
+                tl.setVariable('KubectlOutput', result.toString());
+            }
+
+            // For each output, check if it contains a JSON object
+            result.forEach(res => {
+                if (IsJsonString(res)) {
+                    const jsonResult = JSON.parse(res);
+                    // Check if the output contains a deployment
+                    if (jsonResult.kind && jsonResult.kind.toLowerCase() === deploymentKind) {
+                        // Get all the pods in this cluster
+                        const allPods = JSON.parse(getAllPods(clusterConnection).stdout);
+                        const clusterInfo = getClusterInfo(clusterConnection).stdout;
+                        const metadata = getDeploymentMetadata(jsonResult, allPods, "None", clusterInfo);
+                        const requestUrl = tl.getVariable("System.TeamFoundationCollectionUri") + tl.getVariable("System.TeamProject") + "/_apis/deployment/deploymentdetails?api-version=5.2-preview.1";
+                        sendRequestToImageStore(JSON.stringify(metadata), requestUrl).then((result) => {
+                            tl.debug("DeploymentDetailsApiResponse: " + JSON.stringify(result));
+                        }, (error) => {
+                            tl.warning("publishToImageMetadataStore failed with error: " + error);
+                        });
+                    }
+                }
+            });
+        });
+}
+
+export async function sendRequestToImageStore(requestBody: string, requestUrl: string): Promise<any> {
```

Comment:
NIT: rename to `pushMetadataToArtifactStore` or something like that maybe?

Reply:
Or `pushDeploymentDataToEvidenceStore`

Reply:
Renamed accordingly